### ISSUE TITLE
Token review permission missing when using external vault address

### ISSUE THREAD

**Issue by gokuatkai** [hashicorp/vault-helm#376]
Using an external vault address (aka **external** mode) does not create `ClusterRoleBinding` for `system:auth-delegator` and make the request unable to authenticate:

Pod:
```
2020-08-25T15:14:13.077Z [INFO]  auth.handler: authenticating
2020-08-25T15:14:13.170Z [ERROR] auth.handler: error authenticating: error="Error making API request.

URL: PUT https://my-vault.com:8200/v1/auth/kubernetes/login
Code: 403. Errors:

* permission denied" backoff=2.822450632
```
Server:
```
Aug 25 15:13:10 my-vault.com vault[1803]: 2020-08-25T15:13:10.793Z [ERROR] auth.kubernetes.auth_kubernetes_a3c6e8ec: login unauthorized due to: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"tokenreviews.authentication.k8s.io is forbidden: User \"system:serviceaccount:security:vault-testenv-agent-injector\" cannot create resource \"tokenreviews\" in API group \"authentication.k8s.io\" at the cluster scope","reason":"Forbidden","details":{"group":"authentication.k8s.io","kind":"tokenreviews"},"code":403}
```

**Comment by jasonodonnell** [comment]
Hi @gokuatkai, that's correct, no role is setup for Vault to use.  This would be a nice enhancement to add, though, so labeling this as such.

**Comment by gokuatkai** [comment]
Hi, Do you think we could change `server-clusterrolebinding.yaml`
and add some logic to change the `subjects[].name` to `name: {{ template "vault.fullname" . }}-agent-injector` so it adds automatically the role if mode `external` is specified?

**Comment by gokuatkai** [comment]
so it only consists of a clusterrolebinding and a serviceaccount, maybe it can be a separate chart ?

**Comment by jasonodonnell** [comment]
We already do create SA and CRB, however, that's only triggered when Vault server is installed via Helm.  Basically we can add an additional value to also create those if Vault is external.

**Comment by MPV** [comment]
Has this been solved now (with #392 being merged)?

**Comment by tvoran** [comment]
@MPV Yep, looks like it!

Closing this, #392 was included in v0.8.0: https://github.com/hashicorp/vault-helm/blob/master/CHANGELOG.md#080-october-20th-2020
